flake.nix

```diff
@@ -288,12 +328,25 @@
           src = null;
 
           env = {
+            # Needed for Meson to find Boost.
+            # https://github.com/NixOS/nixpkgs/issues/86131.
+            BOOST_INCLUDEDIR = "${lib.getDev pkgs.boost}/include";
+            BOOST_LIBRARYDIR = "${lib.getLib pkgs.boost}/lib";
             # For `make format`, to work without installing pre-commit
             _NIX_PRE_COMMIT_HOOKS_CONFIG =
               "${(pkgs.formats.yaml { }).generate "pre-commit-config.yaml" modular.pre-commit.settings.rawConfig}";
           };
 
+          mesonFlags =
+            map (transformFlag "libutil") pkgs.nix-util.mesonFlags
+            ++ map (transformFlag "libstore") pkgs.nix-store.mesonFlags
+            ++ lib.optionals havePerl (map (transformFlag "perl") pkgs.nix-perl-bindings.mesonFlags)
```

Comment:
Thoughts on `transformFlag`:
I think `mesonFlags` should be an implementation detail.
Perhaps this is better implemented as a setup hook, or maybe this is just convenient. Not sure.

Reply:
I think eventually in Nixpkgs we should have structured Meson flags, and a nice `shellForPackages` thing that can combine a bunch of things with subprojects and prefix the meson flags of each package.